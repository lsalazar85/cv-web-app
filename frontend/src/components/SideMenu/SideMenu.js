import React from 'react'
import { NavLink } from 'react-router-dom'
import { RiMailSendLine } from 'react-icons/ri'

import {
  VscAccount,
  VscBriefcase,
  VscBook,
  VscCloudDownload,
} from 'react-icons/vsc'

import {
    SideMenuContainer,
    SideIcons,
    SideIconListWrapper,
    SideIconList
} from './styled';

const ICONS = [
  { id: 1, icon: <VscAccount />, url: '/' },
  { id: 2, icon: <VscBook />, url: '/education' },
  { id: 3, icon: <VscBriefcase />, url: '/work-experience' },
  { id: 4, icon: <RiMailSendLine />, url: '/contact' },
]

const SideMenu = () => {
    return(
        <SideMenuContainer>
            <SideIcons position={'center'}>
                <SideIconListWrapper>
                    { ICONS.map(item => (
                        <SideIconList
                            key={item.id}>
                            <NavLink exact to={item.url}>
                                {item.icon}
                            </NavLink>
                        </SideIconList>
                    ))}
                </SideIconListWrapper>
            </SideIcons>
            <SideIcons position={'flex-end'} padddingBotton={32}>
                <VscCloudDownload />
            </SideIcons>
        </SideMenuContainer>
    )
}

export default SideMenu
