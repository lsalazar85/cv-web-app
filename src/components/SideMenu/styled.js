import styled from 'styled-components'

const SideMenuContainer = styled.aside`
    display: flex;
    flex-direction: column;
    padding: 0 32px;
    @media (max-width: 992px){
        border-right: 1px solid #3dba7d;
        padding: 0 15px;
    }
`
const SideIcons = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: ${props => props.position};
    align-items: center;
    height: 100vh;
    color: #ffffff;

    &:last-child{
        padding-bottom: ${props =>  props.position === 'flex-end' && '40px'}
    }

    svg {
        width: 2em;
        height: 2em;
        cursor: pointer;
        color: #ffffff;
        opacity: 0.95;
        transition: 0.25s;

        @media (max-width: 992px){
            width: 1.7em;
            height: 1.7em;
        }
        
        &:hover {
            color: #3dba7d;
            opacity: 1;
        }
    }

    a.active{
        svg {
            color: #3dba7d;
            transform: scale(1.20);
        }
    }

   ul {
        padding-top: 15em;
        @media (max-width: 992px){padding-top: 8em;}

        li {
            margin-bottom: 45px;

            &:last-child{
                margin-bottom: 0px;
            }
        }
   }
`


export { SideMenuContainer, SideIcons }