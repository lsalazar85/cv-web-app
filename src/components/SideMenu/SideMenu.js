import React from 'react';
import {  NavLink } from 'react-router-dom';
import { RiContactsBookUploadLine } from "react-icons/ri";

import { 
    VscAccount, 
    VscBriefcase,
    VscBook,
    VscCloudDownload
} from "react-icons/vsc";

import { SideMenuContainer, SideIcons } from './styled';

const ICONS = [
    { id: 1, icon: <VscAccount />, url:'/'},
    { id: 2, icon: <VscBook />, url:'/education'},
    { id: 3, icon: <VscBriefcase />, url:'/work-experience'},
    { id: 4, icon: <RiContactsBookUploadLine />, url:'/contact'},
]

const SideMenu = () => {
    return(
        <SideMenuContainer>
            <SideIcons position={'center'}>
                <ul>
                    { ICONS.map(item => (
                        <li key={item.id}><NavLink exact to={item.url}>{item.icon}</NavLink></li>
                    ))}
                </ul>
            </SideIcons>
            <SideIcons position={'flex-end'} padddingBotton={32}>
                <VscCloudDownload />
            </SideIcons>
        </SideMenuContainer>
    )
}

export default SideMenu