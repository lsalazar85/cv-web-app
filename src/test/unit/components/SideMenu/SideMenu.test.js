import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { BrowserRouter as Router, Route } from 'react-router-dom';

import SideMenu from '../../../../components/SideMenu';

describe('SideMenu', () => {
    afterEach(cleanup);

    it('should render correctly', () => {
        const { asFragment } = render(
          <Router>
            <Route>
              <SideMenu />
            </Route>
          </Router>
        );
    
        expect(asFragment()).toBeDefined();
        expect(asFragment()).toMatchSnapshot();
    });

})