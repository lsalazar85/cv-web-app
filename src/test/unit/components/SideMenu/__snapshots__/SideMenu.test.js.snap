// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideMenu should render correctly 1`] = `
<DocumentFragment>
  <div>
    SideMenu
  </div>
</DocumentFragment>
`;
